inginious/frontend/plugins/task_hints/static/js/hints_edit.js

```diff
@@ -0,0 +1,164 @@
+let update_hint_id = null;
+
+/* Show a message into an alert */
+function displayAlert(alert_container_id, message, duration){
+    alert = $("#"+alert_container_id);
+    alert.text(message);
+    alert.show();
+    setTimeout(function(){
+        clearAlert(alert_container_id);
+    }, duration);
+}
+
+/* Clear and hide modal alert*/
+function clearAlert(alert_id){
+    let alert = $("#"+alert_id);
+    alert.fadeTo(100, 1).slideUp(250, () => {
+        alert.slideUp(250);
+        alert.text("");
+    });
+}
+
+/* Load code editor for the hint content on modal */
+function loadHintContentCodeEditor(){
+    $("#hint_content_container").show();
+
+    let hints_editor = codeEditors["task_hints_content"];
+    hints_editor.refresh();
+}
+
+/* Save the new/modified hint in table */
+$("#save_hint_button").on('click', function(event){
+
+    event.preventDefault();
+    let hint = $.extend({
+        "title" : "",
+        "penalty" : 0.0,
+        "content" : ""
+    },{
+        "title" : $("#hint_title")[0].value,
+        "penalty" : parseFloat($("#hint_penalty")[0].value).toFixed(1),
+        "content" : $("#hint_content")[0].value
+    });
+
+    /* Check if the entries are allowed on the hint */
+
+    const contentBlankEntry = hint["content"].match(/^\s*$/)
+
+    if(!hint["title"] || contentBlankEntry){
+        const message = "You need to complete mandatory fields";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    }
+    if(isNaN(hint["penalty"])){
+        const message = "Hint penalty needs to be a number";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    } else if (0 > hint["penalty"] || hint["penalty"] > 100){
+        const message = "Hint penalty must be between 0.0% and 100.0%";
+        let editAlert = "hint_edit_alert";
+        displayAlert(editAlert, message, 10000);
+        return ;
+    }
+
+    /* Update the table with the new/modified hint */
+    let new_hint_index = $("#task_hints_table tbody tr").length;
+    if(update_hint_id !== 0 && !update_hint_id){
+        addHintOnTable(hint, new_hint_index);
+    }else{
+        updateHintOnTable(hint, update_hint_id);
```

Comment:
I think the `update_hint_id` should be set to `null` here. What if I update a hint and then add a new hint? this variable will still have the id of the previously edited hint, and thus, the hint new hint is not added.

Reply:
Currently the `update_hint_id` value is set to null when the edit hint modal is closed.